qemu/tests/block_hotplug.py

```diff
@@ -121,42 +126,47 @@ def get_new_disk(disk1, disk2):
 
             device_list.append(device)
             if not new_disks:
-                raise exceptions.TestFail("Cannot find new disk after hotplug.")
+                test.fail("Cannot find new disk after hotplug.")
 
             if params.get("need_plug") == "yes":
                 disk = new_disks[0]
             else:
                 disk = new_disks[num]
 
+            session = vm.wait_for_login(timeout=timeout)
+            if params.get("format_disk", "no") == "yes":
+                if iteration == 0:
+                    error_context.context("Format disk", logging.info)
+                    utils_misc.format_windows_disk(session,
+                                                   disk_index[num],
+                                                   mountpoint=disk_letter[num])
```

Comment:
If it only works for windows, please make the judgement as `if os_type ...`, otherwise, please perform the same action to linux.

Reply:
thanks, has updated~